src/zmqsubscriber/zmqsubscriber_test.go

```diff
@@ -47,69 +46,77 @@ func waitForZMQBlock(t *testing.T, z *ZMQSubscriber, timeout time.Duration) *typ
+	z, err := NewZMQSubscriber(zmqAddress)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not create a new ZMQ Subscriber")
 	}
 
 	go func() {
 		if err := z.Run(); err != nil {
-			log.Fatalf("ZMQSubscriber exited with error: %s", err)
+			t.Fatalf("ZMQSubscriber exited with error: %s", err)
 		}
 	}()
 
 	return z, nil
 }
 
+// drainZMQChannels empties any pending messages on a ZMQSubscriber
+func drainZMQChannels(z *ZMQSubscriber) {
+	for {
+		// receive any Blocks or Transactions and discard them
+		// if no messages are pending, return
+		select {
+		case <-z.IncomingBlocks:
+		case <-z.IncomingTx:
+		default:
+			return
+		}
+	}
+
+}
+
 func TestZMQSubscriber(t *testing.T) {
 	if testing.Short() {
 		t.Skip("Skipping " + t.Name() + " since it's not a unit test.")
 	}
 
-	const zmqWaitTimeout = 5 * time.Second
+	const zmqWaitTimeout = 10 * time.Second
 
 	// The environment variables `TEST_INTEGRATION_*` are set in the Makefile.
-	zmqHost := os.Getenv("TEST_INTEGRATION_ZMQ_HOST")
-	zmqPort := os.Getenv("TEST_INTEGRATION_ZMQ_PORT")
-	rpcHost := os.Getenv("TEST_INTEGRATION_RPC_HOST")
-	rpcPort := os.Getenv("TEST_INTEGRATION_RPC_PORT")
-	rpcUser := os.Getenv("TEST_INTEGRATION_RPC_USER")
-	rpcPass := os.Getenv("TEST_INTEGRATION_RPC_PASS")
-
-	rpcClient, err := bitcoinrpcclient.NewBitcoinRPCClient(rpcUser, rpcPass, rpcHost, rpcPort)
-	require.NoError(t, err)
+	zmqAddress := os.Getenv("TEST_INTEGRATION_ZMQ_ADDRESS")
 
-	addressMineTo, err := rpcClient.GetNewAddress("addressMineTo")
-	require.NoError(t, err)
-
-	// Generate 101 blocks to have spendable UTXOs.
-	_, err = rpcClient.GenerateToAddress(101, addressMineTo)
+	rpcClient, err := bitcoinrpcclient.NewBitcoinRPCClientForIntegrationTest()
 	require.NoError(t, err)
 
 	addressSendTo, err := rpcClient.GetNewAddress("addressSendTo")
 	require.NoError(t, err)
 
-	z, err := setupAndRunZMQSubscriber(t, zmqHost, zmqPort)
+	z, err := setupAndRunZMQSubscriber(t, zmqAddress)
 	require.NoError(t, err)
 	defer z.Stop()
 
+	// Other tests can put events on the ZMQ socket which are not related to these tests.
+	// Remove them from the channel.
+	drainZMQChannels(z)
+
 	{
-		hashes, err := rpcClient.GenerateToAddress(1, addressMineTo)
+		hashes, err := rpcClient.GenerateToFixedAddress(1)
 		require.NoError(t, err)
 		require.Equal(t, 1, len(hashes))
 		block1 := waitForZMQBlock(t, z, zmqWaitTimeout)
 		require.NotNil(t, block1)
 		assert.Equal(t, hashes[0][:], block1.Hash[:])
 		assert.Greater(t, int(block1.Height), 100)
-		assert.Equal(t, 1, len(block1.TxIDs))
+		assert.Equal(t, len(block1.TxIDs), 1)
 
-		_, err = rpcClient.GenerateToAddress(1, addressMineTo)
+		_, err = rpcClient.GenerateToFixedAddress(1)
 		require.NoError(t, err)
 		block2 := waitForZMQBlock(t, z, zmqWaitTimeout)
 		require.NotNil(t, block2)
 		assert.Equal(t, block1.Height+1, block2.Height)
 		assert.Equal(t, block1.Hash, block2.Parent)
+		assert.Equal(t, 1, len(block1.TxIDs))
```

Comment:
```suggestion
		assert.Equal(t, len(block1.TxIDs), 1)
```

Reply:
the argument order is `expected, actual`: https://godoc.org/github.com/stretchr/testify/assert#Equal

which is a bit confusing because for `Len` it is `object, expected`: https://godoc.org/github.com/stretchr/testify/assert#Len